File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IFavePostsView.java
package dev.ragnarok.fenrir.mvp.view;

import java.util.List;

import dev.ragnarok.fenrir.model.Post;
import dev.ragnarok.fenrir.mvp.core.IMvpView;
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView;


public interface IFavePostsView extends IAccountDependencyView, IMvpView, IErrorView, IAttachmentsPlacesView {
    void displayData(List<Post> posts);

    void notifyDataSetChanged();

    void notifyDataAdded(int position, int count);

    void showRefreshing(boolean refreshing);

    void notifyItemChanged(int index);
}

File: app/src/main/kotlin/dev/ragnarok/fenrir/mvp/view/IFavePostsView.kt
package dev.ragnarok.fenrir.mvp.view

import dev.ragnarok.fenrir.model.Post
import dev.ragnarok.fenrir.mvp.core.IMvpView
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView

interface IFavePostsView : IAccountDependencyView, IMvpView, IErrorView, IAttachmentsPlacesView {
    fun displayData(posts: MutableList<Post>)
    fun notifyDataSetChanged()
    fun notifyDataAdded(position: Int, count: Int)
    fun showRefreshing(refreshing: Boolean)
    fun notifyItemChanged(index: Int)
}